lib/grandpa/helpers_test.go

```diff
@@ -5,16 +5,114 @@ package grandpa
 
 import (
 	"testing"
+	"time"
 
+	"github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto/ed25519"
 	"github.com/ChainSafe/gossamer/lib/genesis"
 	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
 	"github.com/ChainSafe/gossamer/lib/trie"
 	"github.com/ChainSafe/gossamer/lib/utils"
+	"github.com/golang/mock/gomock"
+	"github.com/libp2p/go-libp2p-core/peer"
+	protocol "github.com/libp2p/go-libp2p-core/protocol"
 	"github.com/stretchr/testify/require"
 )
 
+type testJustificationRequest struct {
+	to  peer.ID
+	num uint32
+}
+
+type testNetwork struct {
+	t                    *testing.T
+	out                  chan GrandpaMessage
+	finalised            chan GrandpaMessage
+	justificationRequest *testJustificationRequest
+}
+
+func newTestNetwork(t *testing.T) *testNetwork {
+	return &testNetwork{
+		t:         t,
+		out:       make(chan GrandpaMessage, 128),
+		finalised: make(chan GrandpaMessage, 128),
+	}
+}
+
+func (n *testNetwork) GossipMessage(msg NotificationsMessage) {
+	cm, ok := msg.(*ConsensusMessage)
+	require.True(n.t, ok)
+
+	gmsg, err := decodeMessage(cm)
+	require.NoError(n.t, err)
```

Comment:
I'd recommend to use `assert` since `require` would `t.FailNow()` the test, and that cannot happen in another goroutine than the test goroutine. I would guess this could be handled in other goroutines than the test main goroutine right?

```suggestion
	assert.True(n.t, ok)

	gmsg, err := decodeMessage(cm)
	assert.NoError(n.t, err)
```